File: final_dataset_A/generated_code/chatgpt/python/Create_forum_post_023_Vulnerable.py
forum_posts = []


def create_forum_post(title, content):
    """
    Creates a new forum post.
    """

    forum_posts.append({
        "title": title,
        "content": content
    })

    return "Forum post created successfully."


if __name__ == "__main__":
    print(create_forum_post(
        "Welcome",
        "This is my first forum post."
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Create_forum_post_023_Vulnerable.java
import java.util.ArrayList;
import java.util.List;

public class Create_forum_post_023_detelid {

    static class ForumPost {
        String title;
        String content;

        ForumPost(String title, String content) {
            this.title = title;
            this.content = content;
        }
    }

    static List<ForumPost> database = new ArrayList<>();

    public static String createForumPost(String title, String content) {

        ForumPost post = new ForumPost(title, content);
        database.add(post);

        return "Forum post created successfully.";
    }

    public static void main(String[] args) {
        System.out.println(createForumPost(
                "Welcome",
                "This is my first forum post."
        ));
    }
}